targets/TARGET_NUVOTON/TARGET_M480/serial_api.c

```diff
@@ -0,0 +1,1177 @@
+    serial_baud(obj, 9600);
+    // Configure data bits, parity, and stop bits
+    serial_format(obj, 8, ParityNone, 1);
+    
+    obj->serial.vec = var->vec;
+    obj->serial.irq_en = 0;
+    
+#if DEVICE_SERIAL_ASYNCH
+    obj->serial.dma_usage_tx = DMA_USAGE_NEVER;
+    obj->serial.dma_usage_rx = DMA_USAGE_NEVER;
+    obj->serial.event = 0;
+    obj->serial.dma_chn_id_tx = DMA_ERROR_OUT_OF_CHANNELS;
+    obj->serial.dma_chn_id_rx = DMA_ERROR_OUT_OF_CHANNELS;
+#endif
+
+    // For stdio management
+    if (obj->serial.uart == STDIO_UART) {
+        stdio_uart_inited = 1;
+        memcpy(&stdio_uart, obj, sizeof(serial_t));
+    }
+    
+    if (var->ref_cnt) {
+        // Mark this module to be inited.
+        int i = modinit - uart_modinit_tab;
+        uart_modinit_mask |= 1 << i;
+    }
+}
+
+void serial_free(serial_t *obj)
+{
+    const struct nu_modinit_s *modinit = get_modinit(obj->serial.uart, uart_modinit_tab);
+    MBED_ASSERT(modinit != NULL);
+    MBED_ASSERT(modinit->modname == (int) obj->serial.uart);
+    
+    struct nu_uart_var *var = (struct nu_uart_var *) modinit->var;
+    
+    var->ref_cnt --;
+    if (! var->ref_cnt) {
+#if DEVICE_SERIAL_ASYNCH
+        if (obj->serial.dma_chn_id_tx != DMA_ERROR_OUT_OF_CHANNELS) {
+            dma_channel_free(obj->serial.dma_chn_id_tx);
+            obj->serial.dma_chn_id_tx = DMA_ERROR_OUT_OF_CHANNELS;
+        }
+        if (obj->serial.dma_chn_id_rx != DMA_ERROR_OUT_OF_CHANNELS) {
+            dma_channel_free(obj->serial.dma_chn_id_rx);
+            obj->serial.dma_chn_id_rx = DMA_ERROR_OUT_OF_CHANNELS;
+        }
+#endif
+
+        do {
+            UART_Close((UART_T *) NU_MODBASE(obj->serial.uart));
+    
+            UART_DISABLE_INT(((UART_T *) NU_MODBASE(obj->serial.uart)), (UART_INTEN_RDAIEN_Msk | UART_INTEN_THREIEN_Msk | UART_INTEN_RXTOIEN_Msk));
+            NVIC_DisableIRQ(modinit->irq_n);
+    
+            // Disable IP clock
+            CLK_DisableModuleClock(modinit->clkidx);
+        } while (0);
+    }
+    
+    if (var->obj == obj) {
+        var->obj = NULL;
+    }
+    
+    if (obj->serial.uart == STDIO_UART) {
+        stdio_uart_inited = 0;
+    }
+    
+    if (! var->ref_cnt) {
+        // Mark this module to be deinited.
+        int i = modinit - uart_modinit_tab;
+        uart_modinit_mask &= ~(1 << i);
+    }
+}
+
+void serial_baud(serial_t *obj, int baudrate) {
+    // Flush Tx FIFO. Otherwise, output data may get lost on this change.
+    while (! UART_IS_TX_EMPTY((UART_T *) NU_MODBASE(obj->serial.uart)));
+    
+    obj->serial.baudrate = baudrate;
+    UART_Open((UART_T *) NU_MODBASE(obj->serial.uart), baudrate);
+}
+
+void serial_format(serial_t *obj, int data_bits, SerialParity parity, int stop_bits) {
+    // Flush Tx FIFO. Otherwise, output data may get lost on this change.
+    while (! UART_IS_TX_EMPTY((UART_T *) NU_MODBASE(obj->serial.uart)));
+    
+    // TODO: Assert for not supported parity and data bits
```

Comment:
@0xc0170 Please refer to https://github.com/ARMmbed/mbed-os/pull/4608/commits/49977822ff7bd612cc61f4367088e80c14c3e1b1 for the fix.